src/model_training.py

```diff
@@ -312,8 +312,13 @@ def convert_feature_dataframe_to_dict(
 test_gen = DataGenerator(train_data=test_data.limit(10))
 # load input data table as a Spark DataFrame
 input_data = test_gen.training_set.load_df().toPandas()
+# Merge to weather file city index 
+input_data_weather_file_city_index = (
+    input_data
+    .merge(test_gen.weather_features_df[['weather_file_city', 'weather_file_city_index']], on = 'weather_file_city', how = 'left')
+)
 # run prediction and output a N x M matrix of predictions where N is the number of rows in the input data table and M is the number of target columns
-print(model_loaded.predict(input_data))
+print(model_loaded.predict(input_data_weather_file_city_index))
```

Comment:
This is where I was seeing nonseniscal results

Reply:
If you have only run a tiny amount of training data through the fitting than you can't expect reasonable results yet. Have you run a single non-debug run? If so, then you should start to see convergence in the mlflow charts within an hour or so and it should be comparable to the experiments done before this branch was creates. But I think you said you have not been able to run on a GPU. We need to fix that before you can try this.